aesara/link/c/cmodule.py

```diff
@@ -2715,11 +2717,43 @@ def default_blas_ldflags():
         # isort: off
         import numpy.distutils.system_info  # noqa
 
-        # We need to catch warnings as in some cases NumPy print
+        # We need to catch warnings as in some cases NumPy prints
         # stuff that we don't want the user to see.
-        # I'm not able to remove all printed stuff
-        with warnings.catch_warnings(record=True):
-            numpy.distutils.system_info.system_info.verbosity = 0
+        numpy.distutils.system_info.system_info.verbosity = 0  # side-effect
+
+        class NumpyCompatibleStdoutStringIO(io.StringIO):
+            """Temporarily replaces sys.stdout to capture Numpy output.
+
+            Numpy expects the .encoding attribute to be a string, but for io.StringIO
+            it is set to None and not writable, hence the need for this subclass.
+
+            (See forward_bytes_to_stdout in numpy.distutils.exec_command.)
+            """
+
+            encoding = sys.stdout.encoding
+
+        @contextmanager
+        def filter_numpy_missing_executable_warnings():
+            executables = ["g77", "f77", "ifort", "ifl", "f90", "DF", "efl"]
+            with warnings.catch_warnings(record=True):
+                stdout_sio, stderr_sio = NumpyCompatibleStdoutStringIO(), io.StringIO()
+                with redirect_stdout(stdout_sio), redirect_stderr(stderr_sio):
+                    # Body of "with" block executes here:
+                    yield
+                # In case there are messages printed to both stdout and stderr,
+                # they may be incorrectly interspersed. But in the normal case
+                # there should be no output to either, so this should not pose
+                # a problem.
+                for sio, file in ((stdout_sio, sys.stdout), (stderr_sio, sys.stderr)):
+                    lines = sio.getvalue().splitlines()
+                    for line in lines:
+                        if all(
+                            f"Could not locate executable {exe}" not in line
+                            for exe in executables
+                        ):
+                            print(line, file=file)
```

Comment:
Hmm, taking a look at this with fresh eyes I see that it's indeed confusing.

To make the comment more accurate, I think I should say

> In case there are messages ~printed to~ **captured from** both stdout and stderr, they may be **print** incorrectly interspersed. But in the normal case there should be no output to either, so this should not pose a problem.

I should probably also rename `sio` → `captured_buffer` and `file` → `print_dest`...

Reply:
> I'm not entirely clear on what's happening here

Does 0b89bc462d29308b23c823a8edb1b685e420347d help? I improved the variable names and comments.

> it reminds me that we should only suppress specific output messages and not all of them.

Agreed, I opened https://github.com/aesara-devs/aesara/issues/1005 for this